### DISCUSSION TITLE
Please help with (remote) debugging configuration in VScode for Bubble app developing

### DISCUSSION THREAD

**Discussion by abramq** [charmbracelet/bubbletea#1012] [Q&A]
Hello,
I have found here that when we want to debug Bubble app we have to start dlv independently (headless) and connect to it. I am trying it in many ways (as it is described general, with no details), but can't make the debugger work for VScode. All other tutorials I have found are about remote debugging in Docker, no tutorials for local work. Please, can somebody describe step by step how to make Bubble app to be debugged in VScode? Thanks in advance.

**Comment by KevM** [comment]
Checkout my [debug support for VSCode in bubbleo](https://github.com/KevM/bubbleo/tree/main/.vscode) I wish this guidance was better done in the bubbletea readme. I guess we could make a PR. I've suggested it before but didn't get any response.

**Comment by bashbunni** [comment]
Hey @abramq, have you checked out the delve documentation on [how to connect to a headless session in VS Code](https://github.com/golang/vscode-go/blob/master/docs/debugging.md#remote-debugging)? Let me know if that helps to clarify, if so, we can add a link to that in the debugging section. 

VS Code setup
- start headless delve server in a terminal window (command provided in Bubble Tea README)
- create a new debug configuration: "Go: Connect to server"
- match port and host to the server
- run the debugger in VS Code

**Comment by KevM** [comment]
@bashbunni very useful link. It looks like setting up a headless debugging session with `--accept-multiclient` is a one time thing you need to do. 

A nice quality of life improvement with the [tasks.json file](https://github.com/KevM/bubbleo/blob/main/.vscode/tasks.json) I linked to above is that it gives you support for starting a debug session via F5 and stopping again via shift F5. Maybe this is the same has the standalone headless session with the exception that you do not need to start the debugger manually yourself.

**Comment by abramq** [comment]
Thanks for the fast reaction 👍 Well, I was reading those materials before, but after trying more and more I can see now that my (first) problem was with those ${workspaceFolder} and paths. I just do not use workspace directly (or maybe I use it unconsciously - I'm just learning Go :-) ), I have only one folder with *.go file I work with, So, after (totally) removing all that path references from launch.json now I am debugging the application 👍 Anyway... after the debugged application is finishing render of the model, debugger stops working "Unable to process 'stepIn': debuggee is running". No variables, and no stack to view. By the way - at the same time the application is running well in the terminal window. I enclose here a picture of VScode. Maybe this is normal and we can not debug it this way? P.S. I use your example application "Composable Views"[https://github.com/charmbracelet/bubbletea/tree/master/examples/composable-views](url)
![bubbleTea](https://github.com/charmbracelet/bubbletea/assets/25538813/8a404a29-ffaa-420a-98ed-a2de49c68b45)

**Reply by bashbunni** [reply]
Hey, I searched for your error and found a similar issue with the VS Code debugger that may be able to unblock you here. I'm happy to try and reproduce your issue, but it may take a while until I can get around to it. In the meantime, this may help:

https://github.com/golang/vscode-go/issues/2053
